nrf802154_sniffer/nrf802154_sniffer.py

```diff
@@ -84,6 +86,52 @@ def __init__(self):
         self.channel = None
         self.threads = []
 
+        # Time correction variables.
+        self.first_local_timestamp = None
+        self.first_sniffer_timestamp = None
+
+    def correct_time(self, sniffer_timestamp):
+        """
+        Function responsible for correcting the time reported by the sniffer.
+        The sniffer timer has 1us resolution and will overflow after
+        approximately every 72 minutes of operation.
+        For that reason it is necessary to use the local system timer to find the absolute
+        time frame, within which the packet has arrived.
+
+        This method should work as long as the MCU and PC timers don't drift
+        from each other by a value of approximately 36 minutes.
+        :param sniffer_timestamp: sniffer timestamp in microseconds
+        :return: absolute sniffer timestamp in microseconds
+        """
+        if self.first_local_timestamp is None:
+            # First received packets - set the reference time and convert to microseconds.
+            self.first_local_timestamp = int(time.time()*(10**6))
```

Comment:
I think this procedure could be improved even more: each received packet could update `first_local_timestamp` and `first_sniffer_timestamp` variables. It would decrease the problem of sniffer-PC timers drift.

Reply:
Good suggestion. That possibility went over my head.

Reply:
Decided F2F: suggested improvement would complicate this procedure. Without this improvement potential problem occurs after ~2 years of sniffing. Decided to keep it simple.